tests/test_project/CopyModulesWebpackPlugin.test.js

```diff
@@ -86,4 +90,75 @@ describe('copy-modules-webpack-plugin', function() {
 
     testPluginConfig(pluginConfig, pathsThatShouldBeCopied, done);
   });
+
+  it('copies relevant package.json files in addition to javascript files when includePackageJsons is true',
+      function(done) {
+        const pathsThatShouldBeCopied = [
+              'mock_src/b.js',
+              'mock_src/e.js',
+              'mock_src/subfolder/a.js',
+              'package.json',
+              'node_modules/foo-pkg/foo.js',
+              'node_modules/foo-pkg/package.json',
+              'node_modules/bar-pkg/bar.js',
+              'node_modules/bar-pkg/package.json',
+              '__..__/node_modules/outside-cwd-pkg/outside-cwd.js',
+              '__..__/node_modules/outside-cwd-pkg/package.json'
+            ],
+            pluginConfig = { includePackageJsons: true };
+
+        testPluginConfig(pluginConfig, pathsThatShouldBeCopied, done);
+      }
+  );
+
+  it('does not copy package.json files when includePackageJsons is false', function(done) {
+    const pathsThatShouldBeCopied = [
+          'mock_src/b.js',
+          'mock_src/e.js',
+          'mock_src/subfolder/a.js',
+          'node_modules/foo-pkg/foo.js',
+          'node_modules/bar-pkg/bar.js',
+          '__..__/node_modules/outside-cwd-pkg/outside-cwd.js'
+        ],
+        pluginConfig = { includePackageJsons: false };
+
+    testPluginConfig(pluginConfig, pathsThatShouldBeCopied, done);
+  });
+
+  it('handles files that have no parent package.json', function(done) {
```

Comment:
Just curious about your thinking behind this test case - why are you testing this scenario specifically?

Reply:
> Just curious about your thinking behind this test case - why are you testing this scenario specifically?

Because it's possible and because it uniquely tests one of the two reasons why [this loop](https://github.com/sonatype-nexus-community/copy-modules-webpack-plugin/pull/10/files#diff-168726dbe96b3ce427e7fedce31bb0bcR70) would ever exit.  The other tests test the case where the package.json is found and the return statements within the loop are hit, but this test is the only one that checks that the loop condition itself ever goes false.